.github/workflows/haskell.yml

```diff
@@ -235,5 +245,14 @@ jobs:
           export TOOLS_DIR=/opt
           # /usr/local/opt/curl/bin for macOS
           export PATH=$HOME/.local/bin:$HOME/.ghcup/bin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/opt/curl/bin
+          if test -n "$SUBDIR"
+          then
+            mv "$PACKCHECK_LOCAL_PATH" "$SUBDIR/packcheck.sh"
+            # This is required as dist-newstyle will be created where
+            # cabal.project exists in the tree. It is considered the working
+            # directory.
+            echo "packages: streamly-core.cabal" > "$SUBDIR/cabal.project"
```

Comment:
Why do we need a cabal.project for sdist build. ideally we should not use a cabal.project. We can remove the cabal.project in the parent dir instead if packacheck does not have a way to ignore it.

Reply:
Ah, this is what I've done initially.